astropy/io/misc/parquet.py

```diff
@@ -450,7 +450,20 @@ def write_table_parquet(table, output, overwrite=False):
                     val = []
             else:
                 val = encode_table[name]
-            arrays.append(pa.array(val, type=schema.field(name).type))
+            try:
+                arrays.append(pa.array(val, type=schema.field(name).type))
+            except pa.ArrowNotImplementedError as err:
+                # check if val is big-endian.
+                if (np.little_endian and val.dtype.byteorder == ">") or (
+                    not np.little_endian and val.dtype.byteorder == "="
+                ):
+                    # We need to convert the array to little-endian.
+                    val2 = val.byteswap()
```

Comment:
The standard way to do this is `val.astype(new_dtype)`

Note that `val.astype(dtype, copy=False)` is very fast if the `dtype` is identical, so if you know the `dtype` it should become, you can do that upfront, and not have to worry about the `try/except`.

Reply:
Oooh, I didn't realize that.  However, one of the reasons that I did it this way (with the copy) is that I didn't want persisting the table to change the datatype (even endianness) of the columns in the table.  What are your thoughts about that?

Reply:
If you always want a copy, then `.astype` is even better. But I'm not familiar with how `pa.array` works, so cannot really comment on that part. Regardless, I would use `.astype` even for your stanza here, as it is meant to include everything one needs to get to a new dtype, including byte swapping.

Reply:
I don't always want a copy (which would be slower and waste memory), but I do want a copy when we need to do a byte swap because I think it's bad behavior for writing a table to change the datatypes in the table.  So there still needs to be a check for the endianness.  I'll update the code to do an `astype` rather than a `byteswap` and `newbyteorder`.  But there's still a question of whether it should be a try/except (that would only trigger in the hopefully less frequent occasion when a column does need a byte swap) vs checking the byte order of every array every time.

Reply:
Sorry for the confusion `.astype(..., copy=False)` will make a copy if a byteswap is needed, the flag just asks not to copy if it is not needed. So, it should be just right.